zend_abstract_interface/config/config_decode.c

```diff
@@ -127,59 +127,78 @@ static bool zai_config_decode_map(zai_str value, zval *decoded_value, bool persi
                 while (*data && *data != ',') {
                     if (*data == ':') {
                         has_colon = true;
-    
                         data++;
+                        
                         while (*data == ' ' || *data == '\t' || *data == '\n') data++;
 
-                        value_start = data;
-                        value_end = value_start;
-
-                        while (*data && *data != ',') {
-                            if (*data != ' ' && *data != '\t' && *data != '\n') {
-                                value_end = data;
+                        if (*data == ',' || *data == '\0') {
+                            value_start = value_end = NULL;
+                        } else {
+                            value_start = data;
+                            value_end = value_start;
+
+                            while (*data && *data != ',') {
+                                if (*data != ' ' && *data != '\t' && *data != '\n') {
+                                    value_end = data;
+                                }
+                                data++;
                             }
-                            data++;
                         }
 
-                        size_t key_len = key_end - key_start + 1;
-                        size_t value_len = value_end - value_start + 1;
+                        if (key_end && key_start) {
+                            size_t key_len = key_end - key_start + 1;
+                            size_t value_len = (value_end && value_start) ? (value_end - value_start + 1) : 0;
```

Comment:
move this line into the else part below and instead check `if (value_end)`, that's simpler